include/xtensor/xslice.hpp

```diff
@@ -206,6 +215,96 @@ namespace xt
         return xnewaxis_tag();
     }
 
+    template <class A, class B, class C>
+    struct xrange_adaptor
+    {
+        xrange_adaptor(A min, B max, C step)
```

Comment:
``min`` and ``max`` should be avoided as names, like their uppercase counterpart. Windows define ``min`` and ``max`` macros lowercase (yes, I'm serious ...); even if we can prevent their generation with the ``NOMINMAX`` macro, we should avoid possible conflicts.